apps/frontend/src/components/toast/toast.tsx

```diff
@@ -4,67 +4,132 @@ import ClearIcon from '@mui/icons-material/Clear';
 import { Tooltip, Checkbox } from '@mui/material';
 import { useState } from 'react';
 import { ToastModal } from '../toast-modal';
+import { useLoginMutation } from '../../store/services/user.api';
+import {
+  useDeleteToastMutation,
+  useUpdateToastMutation,
+} from '../../store/services/toast.api';
+import {
+  useCreateCriminalMutation,
+  useGetCriminalsQuery,
+} from '../../store/services/criminal.api';
 
 export interface Props {
   name: string;
   date: string;
   reason: string;
-  pastToast: boolean;
+  isPastToast: boolean;
   isUserToast: boolean;
   isConvicting?: boolean;
+  id: string;
+  userId: string;
 }
 export const Toast: React.FC<Props> = ({
   name,
   date,
   reason,
-  pastToast,
+  isPastToast,
   isUserToast,
   isConvicting,
+  id,
+  userId,
 }) => {
   const [isOpen, setIsOpen] = useState(false);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [_, result] = useLoginMutation({
+    fixedCacheKey: 'userKey',
+  });
 
+  const [trigger] = useDeleteToastMutation();
+  const [triggerUpdate] = useUpdateToastMutation();
+  const { data: criminals } = useGetCriminalsQuery();
+  const [triggerCreateCriminal] = useCreateCriminalMutation();
+
+  const updateToastDidHappen = async () => {
+    if (result.data && result.data.isAdmin) {
+      await triggerUpdate({
+        userId: userId,
```

Comment:
```suggestion
        userId,
```
USE SHORTHAND VERSION - APPLY EVERYWHERE